distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -669,19 +669,21 @@ type Table
          - If floating points values are present in the distinct columns, a
            `Floating_Point_Grouping` warning.
     distinct : Vector Text | Column_Selector -> Case_Sensitivity -> Problem_Behavior -> Table
-    distinct self (columns = Column_Selector.By_Name (self.columns.map .name)) case_sensitivity=Case_Sensitivity.Sensitive on_problems=Report_Warning =
+    distinct self (columns = Column_Selector.By_Name (self.columns.map .name)) case_sensitivity=Case_Sensitivity.Default on_problems=Report_Warning =
         problem_builder = Problem_Builder.new
         warning_mapper error = case error of
             No_Output_Columns -> Maybe.Some No_Input_Columns_Selected
             _ -> Nothing
         key_columns = Warning.map_warnings_and_errors warning_mapper <|
             self.columns_helper.select_columns selector=columns reorder=True on_problems=on_problems
-        text_case_insensitive = case case_sensitivity of
-            Case_Sensitivity.Sensitive -> False
+
+        text_case_sensitivity = case case_sensitivity of
             Case_Sensitivity.Insensitive locale ->
                 Helpers.assume_default_locale locale <|
-                    True
-        new_table = self.connection.dialect.prepare_distinct self key_columns text_case_insensitive problem_builder
+                    case_sensitivity
+            _ -> case_sensitivity
```

Comment:
This feels a bit forced, we destructure the `case_sensitivity` parameter just to check if, maybe, the locale provided is wrong and shout 'error'. But we do not transform it at all.

I think with these changes, if the responsibility of handling the True/False/Default peculiarities of case sensitivity were shifted into `dialect.prepare_distinct`, I'd cede the responsibility for verifying the locale there too.

Well, maybe in some far future, some dialect _will_ actually support custom locales, who knows. If we cede the check to dialect it will be customizable. That is not my main argument though, my major point is that here this check looks a bit out of place - there seems to be no reason to do it. Instead I'd make the code in SQLite_Dialect following:

```
if key_column.sql_type.is_definitely_text.not then expr else case text_case_sensitivity of
    Case_Sensitivity.Insensitive locale -> 
       Helpers.assume_default_locale locale <|
           SQL_Expression.Operation "FOLD_CASE" [expr]
    Case_Sensitivity.Sensitive -> SQL_Expression.Operation "MAKE_CASE_SENSITIVE" [expr]
    Case_Sensitivity.Default -> expr
```

IMO this will be much more tidy - as we are now checking the locale at the place where it is _actually_ handled.

Currently, we do this check at the callsite, but maybe we can call `prepare_distinct` from a different site, forgetting about this check - and getting subtle bugs.